tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/horovod_sync_train_test.py

```diff
@@ -50,46 +50,58 @@ def test_adam_minimize_trainable(self):
 
   def common_minimize_trainable(self, base_opt, test_opt, name):
     tf.config.set_soft_device_placement(True)
-    hvd.init()
-    base_opt = de.DynamicEmbeddingOptimizer(base_opt, synchronous=True)
-    for dtype, run_step, dim in itertools.product([dtypes.float32], [1], [10]):
-      x = tf.random.uniform(shape=[32, dim])
-      y = tf.zeros([32, 1])
-
-      global_step = training_util.create_global_step()
-
-      base_weight = tf.compat.v1.get_variable(name="base_weights",
-                                              initializer=tf.ones([10, 1]))
 
-      base_logits = tf.nn.relu(math_ops.matmul(x, base_weight))
-      base_loss = tf.nn.sigmoid_cross_entropy_with_logits(labels=y,
-                                                          logits=base_logits)
-
-      base_opt_op = base_opt.minimize(base_loss,
-                                      global_step,
-                                      var_list=[base_weight])
+    hvd.init()
 
-      test_weight = tf.compat.v1.get_variable(name="test_weights",
-                                              initializer=tf.ones([10, 1]))
+    # These cases need 2 GPUs at least if available.
+    logical_devices = tf.config.list_logical_devices('GPU')
+    _device = "GPU" if len(logical_devices) >= hvd.size() else "CPU"
+    _device_id = hvd.rank(
```

Comment:
Use `hvd.local_rank()` in case someone need to run the tests on multiple nodes.

Reply:
agree

Reply:
Done